core/modules/entity/entity.module

```diff
@@ -43,6 +43,33 @@ function entity_flush_caches() {
   return $bins;
 }
 
+/**
+ * Implements hook_theme().
+ */
+function entity_theme() {
+  // Build a pattern in the form of "(type1|type2|...)(\.|__)" such that all
+  // templates starting with an entity type or named like the entity type
+  // are found.
+  // This has to match the template suggestions provided in
+  // template_preprocess_entity().
+  $types = array();
+  foreach (entity_get_info() as $type => $info) {
+    if (isset($info['controller class']) && in_array('EntityControllerInterface', class_implements($info['controller class']))) {
+      $types[] = $type;
```

Comment:
I suggest: 
```php
      if (!in_array($type, array('node', 'comment', 'file', 'taxonomy_term', 'user'))) {
        $types[] = $type;
      }
``` 
This will leave all 5 core entities out of the pattern, resulting in `template_preprocess_entity` not being called for those, since they provide their own full preprocess functions